src/directives/powerCreeps/baseOperator.ts

```diff
@@ -0,0 +1,163 @@
+import {log} from '../../console/log';
+import {profile} from '../../profiler/decorator';
+import {CombatPlanner, SiegeAnalysis} from '../../strategy/CombatPlanner';
+import {Visualizer} from '../../visuals/Visualizer';
+import {Directive} from '../Directive';
+import {DirectiveNukeResponse} from '../situational/nukeResponse';
+import {GenerateOps} from './powers/generateOps';
+import {Power} from './powers/genericPower';
+import {OperateExtension} from './powers/operateExtension';
+
+
+interface DirectiveBaseOperatorMemory extends FlagMemory {
+	powerPriorities: PowerConstant[];
+}
+
+export enum types {
+	opgen,
+	baseoperator,
+	basedefender
+}
+
+/**
+ * Simple directive to run a power creep where the flag name is the power creep name
+ */
+@profile
+export class DirectiveBaseOperator extends Directive {
+
+	static directiveName = 'BaseOperator';
+	static color = COLOR_CYAN;
+	static secondaryColor = COLOR_PURPLE;
+
+	memory: DirectiveBaseOperatorMemory;
+
+	// Power Creep Hack
+	// powerCreep: PowerCreep;
+	powerCreepName: string;
+
+	defaultPowerPriorities: PowerConstant[] = [
+		PWR_GENERATE_OPS,
+		PWR_REGEN_SOURCE,
+		PWR_OPERATE_TOWER,
+		PWR_OPERATE_STORAGE,
+		PWR_OPERATE_LAB,
+		PWR_OPERATE_SPAWN,
+		PWR_OPERATE_EXTENSION,
+		PWR_REGEN_MINERAL];
+
+	constructor(flag: Flag) {
+		super(flag);
+		const powerCreep = Game.powerCreeps[flag.name];
+		if (!powerCreep) {
+			log.error(`Power Creep not found for ${this.print}, deleting directive`);
+			this.remove();
+		}
+		this.memory.powerPriorities = this.memory.powerPriorities || this.defaultPowerPriorities;
+	}
+
+	spawnMoarOverlords() {
+	}
+
+	init(): void {
+
+	}
+
+
+	// Wrapped powerCreep methods ==================================================================================
+
+	renew(powerCreep: PowerCreep, powerSource: StructurePowerBank | StructurePowerSpawn) {
+		if (powerCreep.pos.inRangeToPos(powerSource.pos, 1)) {
+			return powerCreep.renew(powerSource);
+		} else {
+			return powerCreep.moveTo(powerSource, {ignoreRoads: true, range: 1, swampCost: 1, reusePath: 0,
+				visualizePathStyle: {lineStyle: 'dashed', fill: 'yellow'}});
+		}
+	}
+
+	enablePower(powerCreep: PowerCreep, controller: StructureController) {
```

Comment:
all this should reaaaaaly go on the powerZerg class but we can merge and fix it later

Reply:
Agreed